addon/mixins/run.js

```diff
@@ -367,12 +368,107 @@ export default Mixin.create({
     cancelPoll(label);
   },
 
+  /**
+   Adds a new disposable to the object. A disposable is a function that
+   disposes of bindings that are outside of Ember's lifecyle. This essentially
+   means you can register a function that you want to run to automatically tear
+   down any bindings when the Ember object is destroyed.
+
+   Example:
+
+   ```js
+   // app/components/foo-bar.js
+   import Ember from 'ember';
+   import DOMish from 'some-external-library';
+
+   const { run } = Ember;
+
+   export default Component.extend({
+     init() {
+       this.DOMish = new DOMish();
+
+       this.bindEvents();
+     },
+
+     bindEvents() {
+       let onFoo = run.bind(this.respondToDomEvent);
+       this.DOMish.on('foo', onFoo);
+
+       this.domFooToken = this.registerDisposable(() => this.DOMish.off('foo', onFoo));
+     },
+
+     respondToDOMEvent() {
+       // do something
+     }
+   });
+   ```
+
+   @method registerDisposable
+   @param { Function } disposable
+   @returns A label representing the position of the disposable
+   @public
+   */
+  registerDisposable(disposable) {
+    if (typeof disposable !== 'function') {
+      throw new Error('You must pass a function as a disposable');
+    }
+
+    let disposables = this._getOrAllocateArray('_registeredDisposables');
+
+    return disposables.push(disposable) - 1;
```

Comment:
Chatted with @scalvert a bit about this in slack.  I think we need to modify so that we track the specific items in the token:

```js
registerDisposable(callback) {
  let disposable = {
    disposed: false,
    callback
  };
  
  let disposables = this._getOrAllocateArray('_registeredDisposables');
  disposables.push(disposable);
  return disposable;
},

runDisposable(disposable) {
  if (disposable.disposed) { return; }
  disposable.disposed = true;  
  disposable.callback();
},
```

Reply:
👍  on it